### PR TITLE
Removing Statuses from programs table completely

### LINKED ISSUE #8196: Deprecate the Status_Definitions column in programs table
#### Is your feature request related to a problem? Please describe.

Deprecate the Status_Definitions column in programs table

#### Describe the solution you'd like.

Deprecate the Status_Definitions column in programs table as it's no longer needed. 

#### What priority is this?

P2

Last task to complete #7040

### PR DESCRIPTION
**Pull request by swatkat1**
### Description

Removing Statuses from programs table completely

## Release notes

Removing Statuses from programs table completely

#### General

Read the full guidelines for PRs [here](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#creating-a-pull-request)

- [x] Added the correct label: < feature | enhancement | bug | under-development | dependencies | infrastructure | ignore-for-release | database >
- [x] Assigned to a specific person, `civiform/developers`, or a [more specific round-robin list](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#adding-reviewers)
- [x] Added an additional reviewer from outside your organization as FYI (if the primary reviewer is in the same organization as you)
- [x] Created unit and/or browser tests which fail without the change (if possible)
- [x] Performed manual testing (Chrome and Firefox if it includes front-end changes)
- [x] Extended the README / documentation, if necessary

#### Database evolutions

Read the guidelines [here](https://github.com/civiform/civiform/wiki/Database#writing-database-evolutions)

- [x] Assigned two reviewers
- [x] Guarded against already existing resources using `IF NOT EXISTS` and `IF EXISTS`
- [x] Downs created to undo changes in Ups

### Issue(s) this completes

Fixes #8196
Updated documentation- https://github.com/civiform/civiform/wiki/Database

### CODE REVIEW

File: server/test/repository/ApplicationStatusesRepositoryTest.java
Comment (rockycodes) [inline server/test/repository/ApplicationStatusesRepositoryTest.java:75]: nitpick (non-blocking): Would it work to remove some lines by adding more than one status at a time? Eg. 
```
repo.createOrUpdateStatusDefinitions(
        programName, new StatusDefinitions(ImmutableList.of(REAPPLY_STATUS, APPROVED_STATUS)));
```

File: server/test/repository/ApplicationStatusesRepositoryTest.java
Comment (rockycodes) [inline server/test/repository/ApplicationStatusesRepositoryTest.java:83]: question (non-blocking): If I'm understanding this correctly, we are still adding two statuses like we were previously (REAPPLY and APPROVED). Why are there 4 statuses instead of 3 now?

File: server/test/services/export/AbstractExporterTest.java
Comment (rockycodes) [inline server/test/services/export/AbstractExporterTest.java:86]: nitpick (non-blocking): rename this var `appRepo` or `appStatusRepo` for readability

File: server/test/services/program/ProgramDefinitionTest.java
Comment (rockycodes) [inline server/test/services/program/ProgramDefinitionTest.java:1419]: question (non-blocking): Why remove this?

**Review by rockycodes — APPROVED** [review]
lgtm! hooray for removing all of this! i haven't been following this work very closely, so i mostly have clarifying questions. for example, i'm unclear about when you need to do `repo.createOrUpdateStatusDefinitions` and when you don't.

**Review by rockycodes — CHANGES_REQUESTED** [review]
I just slacked with Nick who is out today and he requested removing the evolution from this PR since it's not hurting anything to have that column in there, and we can remove it in a later PR once we're positive all the relevant data has been migrated and there are no lingering issues.

Update: actually Nick's request is to hold off on merging this until the feature has been sufficiently tested by CEs and we've removed the feature flag.

File: server/test/services/program/ProgramDefinitionTest.java
Comment (swatkat1) [reply server/test/services/program/ProgramDefinitionTest.java:1419]: Statuses are no longer part of Program Definitions. We should not be testing this here.

**Review by rockycodes — APPROVED** [review]
approving this again after our discussion in eng sync. 🎢

File: server/test/actions/ProgramDisabledActionTest.java
Comment (swatkat1) [reply server/test/actions/ProgramDisabledActionTest.java:30]: The sbt-test was complaining about the on unused variables. So misc. cleanup

File: server/test/controllers/admin/AdminProgramControllerTest.java
Comment (swatkat1) [reply server/test/controllers/admin/AdminProgramControllerTest.java:192]: The sbt-test was complaining about the on unused variables. So misc. cleanup

File: server/test/repository/ApplicationStatusesRepositoryTest.java
Comment (dkatzz) [reply server/test/repository/ApplicationStatusesRepositoryTest.java:83]: +1 also curious about this

File: server/test/repository/ApplicationStatusesRepositoryTest.java
Comment (dkatzz) [inline server/test/repository/ApplicationStatusesRepositoryTest.java:86]: **question:** What does `get(1).getStatusDefinitions()
                .getStatuses()
                .get(0)
                .statusText())` return?

File: server/test/services/export/AbstractExporterTest.java
Comment (swatkat1) [reply server/test/services/export/AbstractExporterTest.java:86]: done!

Comment (dkatzz) [inline server/test/services/export/AbstractExporterTest.java:86]: **suggestion:** Can we camel case this? Should be `appRepo` instead of `apprepo`

File: server/test/models/ProgramModelTest.java
Comment (swatkat1) [reply server/test/models/ProgramModelTest.java:100]: Statuses aren't part of Program model anymore. We have other classes to test Statuses now :)

File: server/test/services/program/ProgramDefinitionTest.java
Comment (dkatzz) [inline server/test/services/program/ProgramDefinitionTest.java:1362]: **suggestion:** We should still test this, but with the new status logic to make sure it localizes appropriately. Does that exist somewhere else?

**Review by dkatzz — APPROVED** [review]
A few comments / questions, but overall looks good! Can you confirm that we're testing the localization of statuses somewhere? I also propose we submit this after we confirm everyone is on the version of CiviForm that uses the new statuses.

File: server/test/repository/ApplicationStatusesRepositoryTest.java
Comment (swatkat1) [reply server/test/repository/ApplicationStatusesRepositoryTest.java:83]: I have reached out in the Slack Channel - I am puzzled too

Comment (dkatzz) [reply server/test/repository/ApplicationStatusesRepositoryTest.java:83]: Maybe its worth checking for each whether they're live or obsolete and the name of the status definitions? And / or printing a debug string of `appStatusRepo.lookupAllApplicationStatusesModels( programDefinition.adminName())`?

File: server/test/services/program/ProgramDefinitionTest.java
Comment (swatkat1) [reply server/test/services/program/ProgramDefinitionTest.java:1362]: Localizations are tested as part of StatusService tests.

File: server/test/repository/ApplicationStatusesRepositoryTest.java
Comment (swatkat1) [reply server/test/repository/ApplicationStatusesRepositoryTest.java:75]: no actually the tests is query for all histrorical + active status so that we can use the status text for filtering applications. Technically it doesn't matter if the query returns 3 or 4. We should see both Reapply and Approved status text in the final result. But I am also surprised that the program.build() is getting called twice

File: server/test/controllers/admin/AdminProgramTranslationsControllerTest.java
Comment (swatkat1) [reply server/test/controllers/admin/AdminProgramTranslationsControllerTest.java:381]: I put a small hack. The status gets automatically set on every program creation. I added a line in ProgramBuilder.java to create an empty status for every time any program is built. So, no need to explicitly set for empty status.

File: server/test/repository/ApplicationStatusesRepositoryTest.java
Comment (swatkat1) [reply server/test/repository/ApplicationStatusesRepositoryTest.java:83]: Hi @dkatzz - both are empty. Now I am sure the builder is getting called twice. 

 *********************************************In the builder 1
*********************************************In the builder Current active status -  []
*********************************************In the builder 2
*********************************************In the builder Current active status -  []
********************************************* 2

File: server/test/repository/ApplicationStatusesRepositoryTest.java
Comment (swatkat1) [reply server/test/repository/ApplicationStatusesRepositoryTest.java:78]: fixed it. Thanks @dkatzz . Adding it after the block.build() worked!

File: server/test/repository/ApplicationStatusesRepositoryTest.java
Comment (swatkat1) [reply server/test/repository/ApplicationStatusesRepositoryTest.java:92]: there won't be any statuses as its empty on program creation. I added a line to test it as well provided detail explanation in the unit tests- 
`    assertThat(statusDefinitionsModelResults.get(0).getStatusDefinitions().getStatuses().size())
        .isEqualTo(0);`

File: server/test/services/export/AbstractExporterTest.java
Comment (swatkat1) [reply server/test/services/export/AbstractExporterTest.java]: renamed it :)

**Pull request merged**
